tests/benchdnn/utils/memory.cpp

```diff
@@ -44,24 +44,77 @@ void memory_registry_t::add(void *ptr, size_t size) {
     total_size_ += size;
 
     BENCHDNN_PRINT(8,
-            "[CHECK_MEM]: zmalloc request with size %s, total "
+            "[CHECK_MEM]: zmalloc request (%p) with size %s, total "
             "allocation size: %s\n",
-            smart_bytes(size).c_str(), smart_bytes(total_size_).c_str());
+            ptr, smart_bytes(size).c_str(), smart_bytes(total_size_).c_str());
     warn_size_check();
 }
 
 void memory_registry_t::remove(void *ptr) {
     std::lock_guard<std::mutex> g(m_);
-    const size_t size = allocations_[ptr];
+    // Use `at` to catch cases when unallocated pointers are removed.
+    const size_t size = allocations_.at(ptr);
     total_size_ -= size;
 
     BENCHDNN_PRINT(8,
-            "[CHECK_MEM]: zfree request with size %s, total "
+            "[CHECK_MEM]: zfree request (%p) with size %s, total "
             "allocation size: %s\n",
-            smart_bytes(size).c_str(), smart_bytes(total_size_).c_str());
+            ptr, smart_bytes(size).c_str(), smart_bytes(total_size_).c_str());
     allocations_.erase(ptr);
 }
 
+void memory_registry_t::add_mapped(void *ptr, size_t size) {
+    std::lock_guard<std::mutex> g(m_);
+
+    // When the library memories, which are mapped when created, reordered with
+    // data, they may choose the fast path with CPU reorder. In that case, the
+    // mapped memory pointer for CPU reorder will be used. Since such service
+    // reorders follow the consolidated execution path, they'll unmap those new
+    // memory objects, but since they are not mapped, unmap won't happen. But
+    // after the execution, the mapping happens and returns the same pointer.
+    // Accounting such mapping doesn't make sense as no extra memory is
+    // allocated, thus, report this fact and return.
+    if (mapped_allocations_.find(ptr) != mapped_allocations_.end()) {
+        BENCHDNN_PRINT(8,
+                "[CHECK_MEM]: repeated map request (%p), skip accounting\n",
+                ptr);
+        return;
+    }
+
+    mapped_allocations_.emplace(std::pair<void *, size_t>(ptr, size));
+    total_size_ += size;
+
+    BENCHDNN_PRINT(8,
+            "[CHECK_MEM]: map request (%p) with size %s, total allocation "
+            "size: %s\n",
+            ptr, smart_bytes(size).c_str(), smart_bytes(total_size_).c_str());
+    // Do not warn on overflow as it can be a temporary jump due to reorder or
+    // other internal memory manipulation.
+}
+
+void memory_registry_t::remove_mapped(void *ptr) {
+    std::lock_guard<std::mutex> g(m_);
+
+    // See `add_mapped` comment.
+    // Since double mapping may happen, double mapping may happen as well.
```

Comment:
Missing prefix, it should be: "... double unmaping may happen as well."

Reply:
Thanks for the catch, updated.